testfixtures/tests/test_popen.py

```diff
@@ -106,6 +106,26 @@ def test_read_from_stderr(self):
                 call.Popen('a command', shell=True, stderr=-1, stdout=-1),
                 ], Popen.mock.method_calls)
 
+    def test_read_from_stdout_with_stderr_redirected_check_stderr_in_stdout(self):
+        # setup
+        Popen = MockPopen()
+        Popen.set_command('a command', stdout=b'foo', stderr=b'bar')
+        # usage
+        process = Popen('a command', stdout=PIPE, stderr=STDOUT, shell=True)
+        self.assertTrue(isinstance(process.stdout.fileno(), int))
+        # test stdout contents
+        self.assertIn(b'bar', process.stdout.read())
```

Comment:
Please can you replace these with `compare()` to keep things consistent? (same below)

Reply:
Are you referring to the first or second assertion here?  I can't see how to use compare to replace assertIn.

Reply:
Right, but why not just to a `compare()` on the whole content you expect to be returned from `stdout.read()`?